### PR TITLE
feat(sdk-middleware-logger): support custom logging

### PR DESCRIPTION
**Pull request by VineetKumarKushwaha**
#### support custom logging

Fix: #1632

### CODE REVIEW

File: packages/sdk-middleware-logger/src/logger.js [hunk 1]
```diff
@@ -6,13 +6,19 @@ import type {
...
   return (next: Next): Next => (
     request: MiddlewareRequest,
     response: MiddlewareResponse
   ) => {
     const { error, body, statusCode } = response
     /* eslint-disable */
+    // Pass function to customize logging.
+    if (cb) {
+      cb(request, response)
+    }
```
Comment (daern91) [inline packages/sdk-middleware-logger/src/logger.js:21]: Hmm, this should work but I'm wondering why the customer wouldn't just implement their own logging middleware instead?

File: packages/sdk-middleware-logger/src/logger.js [hunk 2]
```diff
@@ -6,13 +6,19 @@ import type {
...
     console.log('Request: ', request)
     console.log('Response: ', { error, body, statusCode })
```
Comment (daern91) [inline packages/sdk-middleware-logger/src/logger.js:23]: Do they still want these to execute too? Even if they pass in their own logic?

**Review by daern91 — COMMENTED** [review]
Hi Vineet, thanks for the PR. I left some questions below.

File: packages/sdk-middleware-logger/src/logger.js (hunk 1 above)
Comment (jenschude) [reply packages/sdk-middleware-logger/src/logger.js:21]: I think it's still fine to allow custom logging in this middleware, but then I would also create a default callback function which logs to the console

**Review by jenschude — CHANGES_REQUESTED** [review]

**Comment by VineetKumarKushwaha** [comment]
I have resolved the comments.

**Review by emmenko — CHANGES_REQUESTED** [review]
The way it's implemented now does not make much sense to me. You can simply create your own middleware then, which is equivalent to pass a callback function.

**Comment by VineetKumarKushwaha** [comment]
@emmenko this was the requirement so I implemented like that. Could you please suggest me some other approach ?

**Comment by emmenko** [comment]
Well simply create your own custom middleware logger. There is no point in customizing the default logger middleware.

**Comment by emmenko** [comment]
This is the all point of a middleware, you can create your own that fits your needs. See https://commercetools.github.io/nodejs/sdk/Middlewares.html

**Comment by VineetKumarKushwaha** [comment]
@emmenko  could you suggest customer your view here https://jira.commercetools.com/browse/SUPPORT-9830.

**Comment by emmenko** [comment]
I left a comment. You can close this PR.

**Comment by VineetKumarKushwaha** [comment]
@emmenko Thanks

**Pull request closed without merging**